src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -709,14 +816,17 @@ def visit_FunCall(self, node: gtir.FunCall) -> IteratorExpr | ValueExpr:
         elif cpm.is_call_to(node, "neighbors"):
             return self._visit_neighbors(node)
 
+        elif cpm.is_applied_map(node):
+            return self._visit_map(node)
+
         elif cpm.is_applied_reduce(node):
             return self._visit_reduce(node)
 
         elif cpm.is_applied_shift(node):
             return self._visit_shift(node)
 
-        else:
-            assert isinstance(node.fun, gtir.SymRef)
+        elif not isinstance(node.fun, gtir.SymRef):
+            raise NotImplementedError
```

Comment:
I think this function should be split.
In the beginning there is this, let's face it, switch that dispatches the various flavors of a `FunCall` to their respective implementations.
However, the function itself handles the case of `SymRef` that is absolutely unrelated to what the other functions do.
Yo I would suggest moving the code for `SymRef` into its own function and dispatch to it.
It would also make error handling more cleaner.

Reply:
Agreed.